fleetspeak/src/server/mysql/mysql.go

```diff
@@ -69,6 +72,33 @@ func (d *Datastore) runInTx(ctx context.Context, readOnly bool, f func(*sql.Tx)
 	return tx.Commit()
 }
 
+// runInTx runs f in a transaction. If the transaction returns a retriable MySQL
+// error, the transaction will be retried (and f will be run again).
+func (d *Datastore) runInTx(ctx context.Context, readOnly bool, f func(*sql.Tx) error) error {
+	var err error
+L:
+	for i := 0; i < maxRetries; i++ {
+		err = d.runOnce(ctx, readOnly, f)
+		e, ok := err.(*mysql.MySQLError)
+		if !ok {
+			return err
+		}
+		switch e.Number {
+		case 1213, 1205, 1637: // Deadlock, lock timeout, concurrent transactions
+			t := time.NewTimer(time.Duration(i*100) * time.Millisecond)
+			select {
+			case <-t.C:
+				continue L
```

Comment:
As discussed, check if the L is necessary here.

Reply:
done